scripts/firedrake-install

```diff
@@ -488,6 +488,12 @@ pipinstall = pip + ["install",
                     "--no-build-isolation",
                     "--no-binary", ",".join(wheel_blacklist)]
 
+if mode == "install":
+    use_petsc4py_in_petsc = "petsc4py" not in branches
```

Comment:
This remains to support `firedrake-install --doi ...`

Reply:
Please add comment.

Reply:
Added comments.